### PR TITLE
Scoring fixes

### PR DESCRIPTION
**Pull request by OliverW10**
- added a 'hard up' state for scoring that just drive into the wall (needs testing)

### CODE REVIEW

File: controllers/movement.py [hunk 1]
```diff
@@ -192,14 +194,16 @@ def autodrive(self, state_tm: float, initial_call: bool) -> None:
 
         self.time_to_goal = self.trajectory.totalTime() - state_tm
```
Comment (auscompgeek) [inline controllers/movement.py:196]: nit:
```suggestion
    def set_input(self, vx: float, vy: float, vz: float, local: bool, override: bool = False):
```

File: controllers/score_game_piece.py [hunk 2]
```diff
@@ -37,22 +37,32 @@ def driving_to_position(self, initial_call):
         self.movement.set_goal(*get_score_location(self.target_node))
         self.movement.do_autodrive()
         if self.movement.is_at_goal():
-            self.next_state("deploying_arm")
+            self.next_state("hard_up")
+
+    @timed_state(next_state="deploying_arm", duration=0.3, must_finish=True)
+    def hard_up(self):
```
Comment (auscompgeek) [inline controllers/score_game_piece.py:43]:
```suggestion
    def hard_up(self) -> None:
```

File: controllers/score_game_piece.py [hunk 3]
```diff
@@ -37,22 +37,32 @@ def driving_to_position(self, initial_call):
...
+        speed = 0.3
+        vx = speed if is_red() else -speed
+        self.movement.inputs_lock = True
+        self.movement.set_input(vx, 0, 0, False, override=True)
 
     @state(must_finish=True)
-    def deploying_arm(self):
-        self.arm.go_to_setpoint(get_setpoint_from_node(self.target_node))
+    def deploying_arm(self, initial_call):
```
Comment (auscompgeek) [inline controllers/score_game_piece.py:50]:
```suggestion
    def deploying_arm(self, initial_call: bool):
```

File: controllers/score_game_piece.py [hunk 4]
```diff
@@ -37,22 +37,32 @@ def driving_to_position(self, initial_call):
...
         if self.arm.at_goal():
             self.next_state("dropping")
 
-    @state(must_finish=True)
-    def dropping(self):
-        self.gripper.open()
-        if self.gripper.get_full_open():
-            self.done()
+    @timed_state(duration=1, must_finish=True)
+    def dropping(self, initial_call):
```
Comment (auscompgeek) [inline controllers/score_game_piece.py:57]:
```suggestion
    def dropping(self, initial_call: bool):
```

File: physics.py [hunk 5]
```diff
@@ -82,7 +82,7 @@ def __init__(self, physics_controller: PhysicsInterface, robot: MyRobot):
         arm_motors_sim = DCMotor.NEO(2)
         arm_len = (arm.MIN_EXTENSION + arm.MAX_EXTENSION) / 2
         # This value was pulled from the CAD Model on 19/02/23
-        ARM_MASS = 2.305
+        ARM_MASS = 2  # 2.305
```
Comment (auscompgeek) [inline physics.py:85]: Is the comment above still correct?

**Review by auscompgeek — APPROVED** [review]
LGTM with comments

**Review by james-ward — CHANGES_REQUESTED** [review]
The changes to `controllers/arm.py` are clearly breaking the recover controller, as demonstrated on the robot on 27 Feb.

File: controllers/arm.py [hunk 6]
```diff
@@ -79,15 +79,20 @@ class ArmController(StateMachine):
     arm_component: Arm
 
     def __init__(self) -> None:
-        self._target_setpoint: Setpoint = Setpoints.STOW
+        self._target_setpoint: Setpoint = Setpoint(0, 0)
         self._about_to_run: bool = False
+        self.initialised = False
```
Comment (james-ward) [inline controllers/arm.py:84]: Is this needed any more?

File: controllers/arm.py [hunk 7]
```diff
@@ -79,16 +79,18 @@ class ArmController(StateMachine):
...
 
     def go_to_setpoint(self, setpoint: Setpoint) -> None:
         # If this is a different setpoint, we want to take it
         # Interrupt what we are doing and move to the new setpoint
         if setpoint != self._target_setpoint:
             self._about_to_run = True
+            self.next_state("retracting_arm")
             self._target_setpoint = setpoint
-            self.engage("retracting_arm", force=True)
```
Comment (james-ward) [inline controllers/arm.py:91]: If this is the wrong behaviour (I don't think it is) it should be addressed in a separate PR. It's not related to fixing the scoring state machine.

Comment (OliverW10) [reply controllers/arm.py]: Was a mistake made during a rebase, this pr should now not make any changes to the arm

**Review by james-ward — APPROVED** [review]
